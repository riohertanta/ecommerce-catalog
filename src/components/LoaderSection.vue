<!-- template tag used for create reusable components contain a container for all html tag -->
<template>
  <!-- div tag used for parent load spinner css section -->
  <div class="container-load">
    <!-- this tag render load spinner css -->
    <div class="lds-roller">
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
    </div>
  </div>
</template>

<!-- script tag for interactive ecommerce web vue project -->
<script>
export default {
  // name component for export file
  name: "LoaderSection",
};
</script>

<!-- style tag for style web and attribute scoped used for apply to elements of the current component only -->
<style scoped>
@import "../assets/style/LoaderSection.css";
</style>
