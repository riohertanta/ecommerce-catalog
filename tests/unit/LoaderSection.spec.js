import { shallowMount } from "@vue/test-utils";
import LoaderSection from "@/components/LoaderSection.vue";

// describe used to create a block that groups together several related test scenarios
describe("LoaderSection.vue", () => {
  // test used for test scenario to be tested
  test("should have 'LoaderSection' as name", () => {
    // assert
    expect(LoaderSection.name).toBe("LoaderSection");
  });
  test("should render correctly", () => {
    // arrange
    const wrapper = shallowMount(LoaderSection);

    // assert
    expect(wrapper.element).toMatchSnapshot();
  });
});
