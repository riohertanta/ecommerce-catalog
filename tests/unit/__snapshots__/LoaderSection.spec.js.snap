// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderSection.vue should render correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- div tag used for parent load spinner css section -->
  <div
    class="container-load"
  >
    <!-- this tag render load spinner css -->
    <div
      class="lds-roller"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
  
</div>
`;
